src/main/java/roomescape/service/ReservationService.java

```diff
@@ -0,0 +1,67 @@
+package roomescape.service;
+
+import org.springframework.stereotype.Service;
+import roomescape.domain.Reservation;
+import roomescape.domain.ReservationTime;
+import roomescape.dto.SaveReservationRequest;
+import roomescape.dto.SaveReservationTimeRequest;
+import roomescape.repository.ReservationRepository;
+import roomescape.repository.ReservationTimeRepository;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Service
+public class ReservationService {
+    private final ReservationRepository reservationRepository;
+    private final ReservationTimeRepository reservationTimeRepository;
+
+    public ReservationService(
+            final ReservationRepository reservationRepository,
+            final ReservationTimeRepository reservationTimeRepository
+    ) {
+        this.reservationRepository = reservationRepository;
+        this.reservationTimeRepository = reservationTimeRepository;
+    }
+
+    public List<Reservation> getReservations() {
+        return reservationRepository.findAll();
+    }
+
+    public Reservation saveReservation(final SaveReservationRequest request) {
+        return reservationTimeRepository.findById(request.timeId())
+                .map(request::toReservation)
+                .map(reservationRepository::save)
+                .orElseThrow(() -> new NoSuchElementException("해당 id의 예약 시간이 존재하지 않습니다."));
+    }
+
+    public void deleteReservation(final Long reservationId) {
+        checkReservationExist(reservationId);
+        reservationRepository.deleteById(reservationId);
+    }
+
+    private void checkReservationExist(final Long reservationId) {
+        if (!reservationRepository.existById(reservationId)) {
+            throw new NoSuchElementException("해당 id의 예약이 존재하지 않습니다.");
+        }
+    }
+
+    public List<ReservationTime> getReservationTimes() {
+        return reservationTimeRepository.findAll();
+    }
+
+    public ReservationTime saveReservationTime(final SaveReservationTimeRequest request) {
+        return reservationTimeRepository.save(request.toReservationTime());
+    }
+
+    public void deleteReservationTime(final Long reservationTimeId) {
+        checkReservationTimeExist(reservationTimeId);
+        reservationTimeRepository.deleteById(reservationTimeId);
+    }
+
+    private void checkReservationTimeExist(final Long reservationTimeId) {
+        if (!reservationTimeRepository.existById(reservationTimeId)) {
+            throw new NoSuchElementException("해당 id의 예약 시간이 존재하지 않습니다.");
+        }
```

Comment:
```suggestion
        reservationTimeRepository.findById(reservationTimeId)
                .orElseThrow(() -> new NoSuchElementException("해당 id의 예약 시간이 존재하지 않습니다."));
```
Repository에도 남겨둔부분이지만 굳이 exist라는 기능을 만들어야할까싶어요.

Reply:
기존에 이미 존재한 `findById`를 사용해도 괜찮을거 같네요! 적용해보겠습니다!